src/lematerial_forgebench/preprocess/distribution_preprocess.py

```diff
@@ -0,0 +1,73 @@
+from dataclasses import dataclass, field
+
+import numpy as np
+from datasets import load_dataset
+from pymatgen.core import Composition, Element, Structure
+
+from lematerial_forgebench.preprocess.base import BasePreprocessor, PreprocessorConfig
+from lematerial_forgebench.utils.distribution_utils import (
+    map_space_group_to_crystal_system,
+    one_hot_encode_composition,
+)
+
+
+@dataclass
+class DistributionPreprocessorConfig(PreprocessorConfig):
+    name: str
+    description: str
+    n_jobs: int
+
+
+class DistributionPreprocessor(BasePreprocessor):
+    def __init__(
```

Comment:
This should rather use the native `dataclass` attribute definition interface as above 

```python
DistributionPreprocessor(BasePreprocessor):
    distribution_metric: str = "all"
    ...
```

and move the rest of the logic to `def __post_init__(self):`. Interesting pointer on init vs post_init -> https://stackoverflow.com/questions/68439866/python-dataclass-post-initialisation-for-attributes